script/ghci-flags

```diff
@@ -3,63 +3,78 @@
 
 set -e
 
-cd $(dirname "$0")/..
+cd "$(dirname "$0")/.."
 
-root="$(pwd)"
 ghc_version="$(ghc --numeric-version)"
 
-if [ "$1" == "--builddir" ]; then
-  repl_builddir="$2"
-  shift 2
-else
-  repl_builddir=dist-newstyle
-fi
-
-echo "-O0"
-echo "-ignore-dot-ghci"
-
-echo "-outputdir $repl_builddir/build/x86_64-osx/ghc-$ghc_version/semantic-0.8.0.0/build"
-echo "-odir $repl_builddir/build/x86_64-osx/ghc-$ghc_version/semantic-0.8.0.0/build"
-echo "-hidir $repl_builddir/build/x86_64-osx/ghc-$ghc_version/semantic-0.8.0.0/build"
-echo "-stubdir $repl_builddir/build/x86_64-osx/ghc-$ghc_version/semantic-0.8.0.0/build"
-
-echo "-i$root/$repl_builddir/build/x86_64-osx/ghc-$ghc_version/semantic-0.8.0.0/build/autogen"
-
-echo "-I$root/$repl_builddir/build/x86_64-osx/ghc-$ghc_version/semantic-0.8.0.0/build/autogen"
-
-echo "-i$root/semantic-analysis/src"
-echo "-i$root/semantic-ast/src"
-echo "-i$root/semantic-core/src"
-echo "-i$root/semantic-java/src"
-echo "-i$root/semantic-json/src"
-echo "-i$root/semantic-python/src"
-echo "-i$root/semantic-tags/src"
-echo "-i$root/app"
-echo "-i$root/src"
-echo "-i$root/bench"
-echo "-i$root/test"
-
-echo "-optP-include"
-echo "-optP$root/$repl_builddir/build/x86_64-osx/ghc-$ghc_version/semantic-0.8.0.0/build/autogen/cabal_macros.h"
-
-echo "-hide-all-packages"
-
-# Emit package flags from the environment file, removing comments & prefixing with -
-cabal exec --builddir=$repl_builddir -v0 bash -- -c 'cat $GHC_ENVIRONMENT' | grep -v '^--' | sed -e 's/^/-/'
-
-echo "-XHaskell2010"
-echo "-XStrictData"
-
-echo "-Wwarn"
-
-echo "-Weverything"
-echo "-Wno-all-missed-specialisations"
-echo "-Wno-implicit-prelude"
-echo "-Wno-missed-specialisations"
-echo "-Wno-missing-import-lists"
-echo "-Wno-missing-local-signatures"
-echo "-Wno-monomorphism-restriction"
-echo "-Wno-name-shadowing"
-echo "-Wno-safe"
-echo "-Wno-unsafe"
-echo "-Wno-star-is-type"
+# recent hie-bios requires us to output to the file at $HIE_BIOS_OUTPUT, but older builds & script/repl don’t set that var, so we default it to stdout
+output_file="${HIE_BIOS_OUTPUT:-/dev/stdout}"
+
+# do a build of dependencies up front to ensure they’re all available
+cabal v2-build -v0 all --only-dependencies
+
+build_products_dir="dist-newstyle/build/x86_64-osx/ghc-$ghc_version/build-repl"
```

Comment:
1. We only use this directory for `script/repl` since `ghcide` doesn’t do codegen.

2. There’s no reasonable way for us to use the same set of per-component directories as `cabal` will for build products since we can’t reasonably customize this flag based on e.g. the source path.

This means that we’re better off picking a path that `cabal` never will, in order to avoid `cabal build` and `script/repl` attempting to use each others’ interface & object files (cf https://github.com/haskell/cabal/issues/3565 & https://github.com/haskell/cabal/issues/6059) if we e.g. attempt to run `cabal test` from inside `ghci`.

That in turn obviates the need for us to either track the package’s name and version for the path, or to support the use of a non-standard build directory for `script/repl`.

(**NB:** `hie-bios` apparently calls this script with the path to each specific source file as they’re loaded, so we should in principle be able to map those paths back onto specific sets of flags! But `ghci` doesn’t, and since `ghcide` doesn’t build, there’d be no point.)